### ISSUE TITLE
[native-image] memory corruption in byte arrays obtained via getPrimitiveArrayCritical

### ISSUE THREAD

**Issue by johanvos** [oracle/graal#2114]
Rendering a jpeg on linux using JavaFX results in a crash caused by memory corruption.
The function where the corruption happens is this one: 
https://github.com/openjdk/jfx/blob/master/modules/javafx.graphics/src/main/native-iio/jpegloader.c#L1594,

(note: the barray parameter is passed via 0x4, while running this via hotspot just passes the address)

This function has nested calls to ` (*env)->GetPrimitiveArrayCritical`.
Commenting line 1668
         ` jbyte *body = (*env)->GetPrimitiveArrayCritical(env, barray, &iscopy);`
and its corresponding Release method prevents the crash (but obviously data can't be copied into the byte[] in that case, resulting in a black picture.

**Comment by johanvos** [comment]
replacing these 2 lines
```
            jboolean iscopy = FALSE;
            jbyte *body = (*env)->GetPrimitiveArrayCritical(env, barray, &iscopy);
```
with
`            jbyte *body = (*env)->GetPrimitiveArrayCritical(env, barray, NULL);
`
fixes the issue.

**Comment by peter-hofer** [comment]
Thanks for your report! Indeed the pointer incorrectly uses a `CIntPointer`. @mukel has a fix that is in the process of merging.

**Comment by mukel** [comment]
Fixed in f237c4296bae102cd6983ab4f4e0b2f465712a4f
